test/linalg/dense.jl

```diff
@@ -512,6 +512,49 @@ end
     @test_throws ArgumentError diag(zeros(0,1),2)
 end
 
+@testset "New matrix ^ implementation" for elty in (Float64, Complex{Float64})
+    A11 = convert(Matrix{elty}, rand(10, 10))
```

Comment:
You might want to just have `rand(elty,10,10)` here so that you actually make complex matrices.